From 53a2ed4d32facd2e96639cbd3d45f23bf04ef69e Mon Sep 17 00:00:00 2001
From: California Sullivan <california.l.sullivan@intel.com>
Date: Mon, 14 Oct 2019 16:50:19 -0700
Subject: [PATCH] Makefile: propagate CFLAGS

I'm not sure why this only starts happening with v1.9.2, but CFLAGS
aren't carrying over to the program and lib builds with 32-bit,
resulting in the following:

make[1]: Entering directory '/builddir/build/BUILD/build32/programs'
gcc -O3   -I../lib -DXXH_NAMESPACE=LZ4_  -c -o bench.o bench.c
gcc -O3   -I../lib -DXXH_NAMESPACE=LZ4_  -c -o datagen.o datagen.c
gcc -O3   -I../lib -DXXH_NAMESPACE=LZ4_  -c -o lz4cli.o lz4cli.c
gcc -O3   -I../lib -DXXH_NAMESPACE=LZ4_  -c -o lz4io.o lz4io.c
gcc -O3   -I../lib -DXXH_NAMESPACE=LZ4_ -m32 -mstackrealign ../lib/lz4.o ../lib/lz4frame.o ../lib/lz4hc.o ../lib/xxhash.o bench.o datagen.o lz4cli.o lz4io.o -o lz4
/usr/bin/ld: i386:x86-64 architecture of input file `../lib/lz4.o' is incompatible with i386 output
/usr/bin/ld: i386:x86-64 architecture of input file `../lib/lz4frame.o' is incompatible with i386 output
/usr/bin/ld: i386:x86-64 architecture of input file `../lib/lz4hc.o' is incompatible with i386 output
/usr/bin/ld: i386:x86-64 architecture of input file `../lib/xxhash.o' is incompatible with i386 output
/usr/bin/ld: i386:x86-64 architecture of input file `bench.o' is incompatible with i386 output
/usr/bin/ld: i386:x86-64 architecture of input file `datagen.o' is incompatible with i386 output
/usr/bin/ld: i386:x86-64 architecture of input file `lz4cli.o' is incompatible with i386 output
/usr/bin/ld: i386:x86-64 architecture of input file `lz4io.o' is incompatible with i386 output

Shove the CFLAGS into these make calls to fix it.

Signed-off-by: California Sullivan <california.l.sullivan@intel.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index f25f951..50b74c4 100644
--- a/Makefile
+++ b/Makefile
@@ -50,13 +50,13 @@ allmost: lib lz4
 .PHONY: lib lib-release liblz4.a
 lib: liblz4.a
 lib lib-release liblz4.a:
-	@$(MAKE) -C $(LZ4DIR) $@
+	@$(MAKE) -C $(LZ4DIR) CFLAGS="$(CFLAGS)" $@
 
 .PHONY: lz4 lz4-release
 lz4 : liblz4.a
 lz4-release : lib-release
 lz4 lz4-release :
-	@$(MAKE) -C $(PRGDIR) $@
+	@$(MAKE) -C $(PRGDIR) CFLAGS="$(CFLAGS)" $@
 	@cp $(PRGDIR)/lz4$(EXT) .
 
 .PHONY: examples
-- 
2.23.0

